libs/etterlatte-ktor/src/main/kotlin/RestModule.kt

```diff
@@ -26,14 +31,68 @@ import io.ktor.server.routing.IgnoreTrailingSlash
 import io.ktor.server.routing.Route
 import io.ktor.server.routing.route
 import io.ktor.server.routing.routing
+import io.ktor.util.pipeline.PipelinePhase
 import no.nav.etterlatte.libs.common.logging.CORRELATION_ID
 import no.nav.etterlatte.libs.common.objectMapper
 import no.nav.security.token.support.v2.TokenValidationContextPrincipal
 import no.nav.security.token.support.v2.tokenValidationSupport
 import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import org.slf4j.event.Level
 import java.util.*
 
+class PluginConfiguration {
+    var ressursHarAdressebeskyttelse: (id: String) -> Boolean = { false }
+}
+
+private object AdressebeskyttelseHook : Hook<suspend (ApplicationCall) -> Unit> {
+    private val AdressebeskyttelseHook: PipelinePhase = PipelinePhase("Adressebeskyttelse")
+    private val AuthenticatePhase: PipelinePhase = PipelinePhase("Authenticate")
+    override fun install(
+        pipeline: ApplicationCallPipeline,
+        handler: suspend (ApplicationCall) -> Unit
+    ) {
+        /*
+        Skal bli gjort automatisk i authentication plugin, men ligger ikke i pipelinen her
+        selvom den blir lagt til i AuthenticationHook pipeline.insertPhaseAfter(ApplicationCallPipeline.Plugins, AuthenticatePhase)
+         */
+        pipeline.insertPhaseAfter(ApplicationCallPipeline.Plugins, AuthenticatePhase)
```

Comment:
Denne er litt weird, for å slippe dette kan man feks bruke den predefinerte hooken `AuthenticationChecked`, men skal man lage en hook per steg feks adressebeskyttelse, egen ansatt, enhetsbasert og les/skriv(KS) så får man problemet med reordering.
Greia er at  pipeline.items(phases) ikke inneholder `PipelinePhase("Authenticate")` slik den defineres i `AuthenticationHook`. Selvom vi har en install på den fasen via `authenticate{...}`.